analyze/analyses/dominators/mod.rs

```diff
@@ -6,15 +6,21 @@ use twiggy_ir as ir;
 use twiggy_opt as opt;
 use twiggy_traits as traits;
 
-use crate::analyses::utils;
+use crate::analyses::garbage;
 
 mod emit;
 
 struct DominatorTree {
     tree: BTreeMap<ir::Id, Vec<ir::Id>>,
     items: Vec<ir::Id>,
-    unreachable_items: Vec<ir::Id>,
     opts: opt::Dominators,
+    unreachable_items_summary: Option<UnreachableItemsSummary>,
+}
+
+struct UnreachableItemsSummary {
```

Comment:
I wrapped this up into a struct, because the size, size percentage, and number of unreachable items are conditionally shown in the output. This seemed more elegant & less error prone than multiple `Option<_>` members, and we can destructure this with an `if let Some(..)` in the emitting functions

Reply:
&& good idea re: moving this computation up, it makes the `emit_*` methods much cleaner :smile_cat: